public/app/plugins/datasource/graphite/datasource.ts

```diff
@@ -326,13 +360,41 @@ export class GraphiteDatasource
     return expandedQueries;
   }
 
-  annotationQuery(options: any) {
-    // Graphite metric as annotation
-    if (options.annotation.target) {
-      const target = this.templateSrv.replace(options.annotation.target, {}, 'glob');
+  annotationEvents(range: any, target: GraphiteQuery) {
+    if (target.eventsQuery?.tags) {
```

Comment:
Looks like a regression. The original behaviour if both target and tags set, target with take precedence over tags.

Reply:
This makes sense, thank you. If someone had added an annotation with both a target and tags, my old code would have broken their annotation because it would have read the tags instead, causing the regression. I switched it so that it checks for target first to fix the regression.